ITE: Fix the leakage issue when GPE1/GPE2 is used as alternate function

GTLin08 opened the pull request:
This PR will fix the leakage issue when GPE1/GPE2 is used as I2C
The commits include:
1. Modify the sequence of pinctrl setting
2. Disable EGAD pin output of external gpio control
3. Besides, extended setting required to use i2c5

GTLin08 committed:
ITE: drivers/i2c: Extended setting required to use i2c5

In addition to setting SMB5PS, PMER1 also needs to be set when
using i2c5.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

keith-zephyr commented on dts/riscv/ite/it82xx2.dtsi:
```diff
@@ -521,9 +521,9 @@
 			compatible = "ite,it8xxx2-pinctrl-func";
 			reg = <0x00f01680 8>;   /* GPCR */
-			func3-gcr =     <0xf02032 0xf03e16 0xf03e16 NO_FUNC
+			func3-gcr =     <0xf02032 0xf03e16 0xf02032 NO_FUNC
```
This the PMER1 register, correct? Can you use the `func3-ext` property here instead to set PMER1 to the correct value?

GTLin08 replied:
Yes. Done using `func3-ext`

Dino-Li committed:
drivers/pinctrl: it8xxx2: Modify the sequence of pinctrl setting

Modify the sequence of pinctrl setting. Default setting as input mode
prevents leakage during changes to extended setting.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>
Signed-off-by: Dino Li <Dino.Li@ite.com.tw>

Dino-Li committed:
soc: it8xxx2: Disable EGAD pin output of external gpio control

Setting IT8XXX2_EGPIO_EEPODD bit will disable EGAD pin output driving
to avoid leakage when GPIO E1/E2 on it82002 are set to alternate
function.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>
Signed-off-by: Dino Li <Dino.Li@ite.com.tw>

keith-zephyr commented on dts/riscv/ite/it82xx2.dtsi:
```diff
@@ -525,6 +525,10 @@
 					 NO_FUNC  0xf03e10 NO_FUNC  0xf02032>;
 			func3-en-mask = <0x01     0x20     0x20     0
 					 0        0x08     0        0x01    >;
+			func3-ext =     <NO_FUNC 0xf02032  0xf02032 NO_FUNC
```
As a follow-up can you move GPE0 and GPE7 setting from func3-gcr to func3-ext?

GTLin08 replied:
OK, I will create another PR.
